main.cpp

```diff
@@ -12,6 +12,10 @@
 
 void dump_tokens(bf::lex::lexer& l);
 void dump_commands(const bf::prog::program& p);
+inline void dump_help(const char **argv) {
```

Comment:
Can this be just a forward decl?  Let's move the decl above `dump_tokens` and it's implementation likewise above `dump_tokens` impl.